src/Pages/SignUpFormik.tsx

```diff
@@ -29,102 +29,109 @@ const theme = createTheme();
-    }
+  // if (companyName === undefined) {
+  //   alert('invalid Company Name');
+  //   return;
+  // }
 
-    const lastLogin = 'null';
-    const acceptedTermAndConditions = false;
+  // const lastLogin = 'null';
+  // const acceptedTermAndConditions = false;
 
-    const account = {
-      _id: undefined,
-      emailAddress,
-      isAdmin,
-      password,
-      firstName,
-      lastName,
-      companyName,
-      lastLogin,
-      acceptedTermAndConditions,
-    };
-
-    const validationSchema = yup.object({
-      emailAddress: yup
-        .string()
-        .email('Bitte vollständige E-Mail-Adresse eingeben!')
-        .required('E-Mail-Adresse ist erforderlich.'),
-      name: yup
-        .string()
-        .min(3, 'Name should be of minimum 3 characters length')
-        .required('Name ist erforderlich.'),
-    });
+  // const account = {
+  //   _id: undefined,
+  //   emailAddress,
+  //   isAdmin,
+  //   password,
+  //   firstName,
+  //   lastName,
+  //   companyName,
+  //   lastLogin,
+  //   acceptedTermAndConditions,
+  // };
+  // };
+  const validationSchema = yup.object({
+    emailAddress: yup
+      .string()
+      .email('Bitte vollständige E-Mail-Adresse eingeben!')
+      .required('E-Mail-Adresse ist erforderlich.'),
+    password: yup
+      .string()
+      .min(8, 'Passwort sollte mindestens eine Länge von 8 Zeichen haben.')
+      .required('Passwort ist erforderlich.'),
+    firstName: yup
+      .string()
+      .min(2, 'Vorname sollte mindestens eine Länge von 2 Zeichen haben.')
+      .required('Vorname ist erforderlich.'),
+    lastName: yup
+      .string()
+      .min(2, 'Nachname sollte mindestens eine Länge von 2 Zeichen haben.')
+      .required('Nachname ist erforderlich.'),
+    companyName: yup
+      .string()
+      .required('Der Name des Unternehmens ist erforderlich.'),
+  });
 
-    const formik = useFormik({
-      initialValues: {
-        _id: undefined,
-        emailAddress: '',
-        isAdmin: '',
-        password: '',
-        firstName: '',
-        lastName: '',
-        companyName: '',
-        lastLogin: '',
-        acceptedTermAndConditions: false,
-      },
-      validationSchema: validationSchema,
-      onSubmit: (values) => {
-        CreateAccount(account)
-          .then((account) => {
-            // alert('success');
-            // success
-            setCookie('email', emailAddress, { path: '/' });
-            return navigate('/my-products');
-          })
-          .catch((msg) => {
-            alert('error');
-            // error
-          });
-      },
-    });
-  };
+  const formik = useFormik({
+    initialValues: {
+      _id: undefined,
+      emailAddress: '',
+      isAdmin: '',
+      password: '',
+      firstName: '',
+      lastName: '',
+      companyName: '',
+      lastLogin: '',
+      acceptedTermAndConditions: false,
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values) => {
+      CreateAccount(values)
```

Comment:
Have a look at your [AccountType](https://github.com/mhoehne/ecozer.io/blob/main/src/API.ts#L14-L24) and then at your error:

> Argument of type '{ _id: undefined; emailAddress: string; **isAdmin: string**; password: string; firstName: string; lastName: string; companyName: string; lastLogin: string; acceptedTermAndConditions: boolean; }' is not assignable to parameter of type 'AccountType'.

Some of your field types don't match. Look at `isAdmin`, its return type and which type it has before attempting to create the account.

Reply:
ah, got it. stupid me haha